{{cookiecutter.module_name}}/main.py

```diff
@@ -1,94 +1,74 @@
-# input queue (in the case of this sample, "input1").  Because this is a filter module, 
-# we will forward this message onto the "output1" queue.
-def receive_message_callback(message, hubManager):
-    global RECEIVE_CALLBACKS
-    message_buffer = message.get_bytearray()
-    size = len(message_buffer)
-    print ( "    Data: <<<%s>>> & Size=%d" % (message_buffer[:size].decode('utf-8'), size) )
-    map_properties = message.properties()
-    key_value_pair = map_properties.get_internals()
-    print ( "    Properties: %s" % key_value_pair )
-    RECEIVE_CALLBACKS += 1
-    print ( "    Total calls received: %d" % RECEIVE_CALLBACKS )
-    hubManager.forward_event_to_output("output1", message, 0)
-    return IoTHubMessageDispositionResult.ACCEPTED
-
-
-class HubManager(object):
-
-    def __init__(
-            self,
-            protocol=IoTHubTransportProvider.MQTT):
-        self.client_protocol = protocol
-        self.client = IoTHubModuleClient()
-        self.client.create_from_environment(protocol)
-
-        # set the time until a message times out
-        self.client.set_option("messageTimeout", MESSAGE_TIMEOUT)
-        
-        # sets the callback when a message arrives on "input1" queue.  Messages sent to 
-        # other inputs or to the default will be silently discarded.
-        self.client.set_message_callback("input1", receive_message_callback, self)
-
-    # Forwards the message received onto the next stage in the process.
-    def forward_event_to_output(self, outputQueueName, event, send_context):
-        self.client.send_event_async(
-            outputQueueName, event, send_confirmation_callback, send_context)
-
-def main(protocol):
-    try:
-        print ( "\nPython %s\n" % sys.version )
-        print ( "IoT Hub Client for Python" )
-
-        hub_manager = HubManager(protocol)
-
-        print ( "Starting the IoT Hub Python sample using protocol %s..." % hub_manager.client_protocol )
-        print ( "The sample is now waiting for messages and will indefinitely.  Press Ctrl-C to exit. ")
-
-        while True:
-            time.sleep(1)
-
-    except IoTHubError as iothub_error:
-        print ( "Unexpected error %s from IoTHub" % iothub_error )
-        return
-    except KeyboardInterrupt:
-        print ( "IoTHubModuleClient sample stopped" )
-
-if __name__ == '__main__':
-    main(PROTOCOL)
\ No newline at end of file
+# Copyright (c) Microsoft. All rights reserved.
+# Licensed under the MIT license. See LICENSE file in the project root for
+# full license information.
+
+import time
+import os
+import sys
+import asyncio
+from six.moves import input
+import threading
+from azure.iot.device.aio import IoTHubModuleClient
+
+async def main():
+    try:
+        print ( "\nPython %s\n" % sys.version )
+        print ( "IoT Hub Client for Python" )
+
+        print ( "The sample is now waiting for messages and will indefinitely. ")

```

Comment:
will indefinitely?

Reply:
Followed existing template. Will update the wording and move to latter place.